CMakeLists.txt

```diff
@@ -0,0 +1,213 @@
+cmake_minimum_required(VERSION 3.12)
+cmake_policy(SET CMP0076 NEW)
+
+list(APPEND CMAKE_MODULE_PATH "${CMAKE_SOURCE_DIR}/sys/cmake")
+include(CMakeDependentOption)
+include(ucm)
+include(PrecompiledHeader)
+include(SvnVersion)
+
+project(TheDarkMod VERSION 2.08)
+add_executable(TheDarkMod)
+
+cmake_dependent_option(WITH_TOOLS "Compile with Tools" ON "MSVC" OFF)
+set(GAME_DIR "${PROJECT_SOURCE_DIR}/../darkmod" CACHE STRING "The directory where the game assets are stored")
+option(COPY_EXE "Copy executable to GAME_DIR after build" ON)
+
+set(CMAKE_CXX_STANDARD 14)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+set(CMAKE_CXX_EXTENSIONS OFF)
+
+get_svn_revision(SVNVERSION)
+configure_file("idlib/svnversion.cmake.h" "${PROJECT_SOURCE_DIR}/idlib/svnversion.h" @ONLY)
+
+# gather list of source files
+ucm_add_dirs(
+		"cm"
+		"framework"
+		"game"
+		"idlib"
+		"renderer"
+		"sound"
+		"tests"
+		"TypeInfo"
+		"ui"
+		"tools/compilers"
+		TO TDM_SOURCE_FILES RECURSIVE
+)
+ucm_add_files("sys/sys_local.cpp" "sys/sys_local.h" "sys/sys_public.h" TO TDM_SOURCE_FILES)
+
+if (WITH_TOOLS)
+    ucm_add_dirs("tools" TO TDM_SOURCE_FILES RECURSIVE)
+	ucm_remove_files(
+			"tools/edit_stub.cpp"
+			"tools/decl/DialogEntityDefEditor.cpp"
+			"tools/common/RenderBumpFlatDialog.cpp"
+			"tools/guied/GEWindowWrapper_stub.cpp"
+			FROM TDM_SOURCE_FILES
+	)
+else()
+	ucm_add_files("tools/guied/GEWindowWrapper_stub.cpp" TO TDM_SOURCE_FILES)
+	add_compile_definitions(NO_MFC)
+endif()
+
+if (UNIX)
+	ucm_add_dirs("sys/linux" "sys/posix" TO TDM_SOURCE_FILES RECURSIVE)
+    ucm_remove_directories("idlib/sys/win32" FROM TDM_SOURCE_FILES)
+	ucm_remove_files(
+			"renderer/glad_wgl.c"
+			"sys/linux/dedicated.cpp"
+			"sys/linux/test_scheduler.c"
+			"sys/posix/platform_osx.cpp"
+			FROM TDM_SOURCE_FILES)
+elseif (WIN32)
+	ucm_add_dirs("sys/win32" TO TDM_SOURCE_FILES RECURSIVE)
+	ucm_remove_directories("idlib/sys/posix" FROM TDM_SOURCE_FILES)
+	ucm_remove_files("renderer/glad_glx.c" "tools/compilers/dmap/optimize_gcc.cpp" "idlib/bv/Frustum_gcc.cpp" FROM TDM_SOURCE_FILES)
+endif()
+
+# blacklist dead code files that unfortunately still hang around the repository
+ucm_remove_directories("framework/async" FROM TDM_SOURCE_FILES)
+ucm_remove_files(
+		"TypeInfo/main.cpp"
+		"game/Game_network.cpp"
+		"game/MultiplayerGame.cpp"
+		"game/gamesys/Callbacks.cpp"
+		"idlib/math/Simd_AltiVec.cpp"
+		"renderer/draw_exp.cpp"
+        "sys/win32/win_gamma.cpp"
+		FROM TDM_SOURCE_FILES
+)
+
+target_sources(TheDarkMod PRIVATE ${TDM_SOURCE_FILES})
+
+if (MSVC)
+	ucm_add_dirs("sys/msvc/natvis" TO TDM_NATVIS_FILES RECURSIVE ADDITIONAL_EXT "natvis")
+	target_sources(TheDarkMod INTERFACE ${TDM_NATVIS_FILES})
+endif()
+
+# configure precompiled header
+# note TypeInfo.cpp must be excluded from precompiled header because it uses some dirty hacks
+add_precompiled_header(TheDarkMod "idlib/precompiled.h" "idlib/precompiled.cpp" "precompiled.h"
```

Comment:
Does not work for me. Compiler complains that some language constructs require at least C++11.

Note that you have set C++14 above, and that my compiler (GCC 5.4.0) seems to support it.
Maybe the precompiler header macro does not propagate C++ standard setting properly...

Reply:
Same with this line commented out.
Somehow, C++14 is dropped from compile flags.

Reply:
Adding `add_definitions(-std=c++14)` fixes the problem, as long as I keep PCH disabled. If I enable PCH, it does not compile complaining about the lack of c++11. But the hell is wrong with me?...

Reply:
It's probably either the CMake or GCC version which behaves differently than the ones I've tested with. Just for reference, what version of CMake are you using? If it's not 3.15, could you (easily) test with 3.15 just to rule that out?

Reply:
I'm using GCC 5.4.0, which is stock on Ubuntu 16.04, and CMake 3.15.5, which I had to build myself.

Reply:
I guess I have found the problem: you perform add_executable too early. The cxx_standard is specified later, and it seems that it does not affect TheDarkMod executable. To be honest, it would be rather strange if setting a global variable affected the projects defined later... although this is cmake =) Could you please clarify which order of statements should be used to make everything work?

I have tried to put add_executable immediately before you add source files to it, and in that case C++14 got assigned normally (without hacky add_definitions). However, as soon as I uncomment precompiled header, it stops working again --- PCH also does not catch the c++14 flag.

Reply:
Thanks for your analysis. I guess newer GCC versions might derive C++14 from the project somehow, or perhaps it's just the default.
I'll try and hack the C++14 flags into the precomnpiled header options somehow.

Reply:
I've added `-std=c++14` explicitly to the compile options in general and for the precompiled header. My assumption is that this setting is default in later versions, but not yours, and for some reason CMake doesn't propagate it properly. But unfortunately, I can't test it because the oldest GCC version I have access to right now is 6, and that just works. Would appreciate it if you gave it another quick spin. If that doesn't fix it, we may have to disable PCH for older GCC versions...

Reply:
Yes, it seems that GCC 6 has enabled at least C++11 by default.
You can add `constexpr if` into some common header and test it with C++17 standard =)

Reply:
I've excluded Mover.cpp from the precompiled header targets. Could you check if that solves it? I have no idea what's going on, if that doesn't fix it, we may have to disable PCH for GCC below 6.

Reply:
I have fixed the problem in svn rev 8424. You can revert this "exclusion" back.